packages/core/solid/use-chat.ts

```diff
@@ -84,11 +85,23 @@ export function useChat({
   body,
   streamMode,
   generateId = generateIdFunc,
-}: UseChatOptions = {}): UseChatHelpers {
+}: Omit<UseChatOptions, 'api'> & {
+  api?: string | Accessor<string>;
+} = {}): UseChatHelpers & {
+  experimental_addToolResult: ({
+    toolCallId,
+    result,
+  }: {
+    toolCallId: string;
+    result: any;
+  }) => void;
+} {
   // Generate a unique ID for the chat if not provided.
   const chatId = id || `chat-${uniqueId++}`;
 
-  const key = `${api}|${chatId}`;
+  const getApi = () => (typeof api === 'string' ? api : api());
+
+  const key = `${getApi()}|${chatId}`;
```

Comment:
The original is better

Reply:
This code is designed to make the useChat hook reactive so that any changes in the API state are automatically reflected in the `useChat` hook. In SolidJS, state primitives (signals) are functions that need to be called to retrieve the latest value of the API state. In contrast, React allows you to directly read the API state to get its latest value. This fundamental difference necessitated my changes.

```js
const [api, setApi] = createSignal();
const chat = useChat({ api }); // Here, api is a function, hence it will be reactive.
```

Reply:
thanks, that makes sense!